be/src/storage/lake/tablet_manager.cpp

```diff
@@ -670,6 +657,29 @@ StatusOr<TabletSchemaPtr> TabletManager::get_tablet_schema(int64_t tablet_id, in
     return schema;
 }
 
+StatusOr<TabletSchemaPtr> TabletManager::get_tablet_schema_by_index_id(int64_t tablet_id, int64_t index_id) {
+    auto global_cache_key = global_schema_cache_key(index_id);
+    auto schema = lookup_tablet_schema(global_cache_key);
+    TEST_SYNC_POINT_CALLBACK("get_tablet_schema_by_index_id.1", &schema);
+    if (schema != nullptr) {
+        return schema;
+    }
+    // else: Cache miss, read the schema file
+    auto schema_file_path = join_path(tablet_root_location(tablet_id), schema_filename(index_id));
+    auto schema_or = load_and_parse_schema_file(schema_file_path);
+    TEST_SYNC_POINT_CALLBACK("get_tablet_schema_by_index_id.2", &schema_or);
+    if (schema_or.ok()) {
+        VLOG(3) << "Got tablet schema of id " << index_id << " for tablet " << tablet_id;
+        schema = std::move(schema_or).value();
+        // Save the schema into the in-memory cache, use the schema id as the cache key
+        auto cache_value = std::make_unique<CacheValue>(schema);
+        fill_metacache(global_cache_key, cache_value.release(), 0);
```

Comment:
why the cache value size is 0?

Reply:
The schema object is managed by the GlobalTabletSchemaMap

Reply:
line 656 and 957
auto cache_size = `inserted ? (int)schema->mem_usage() : 0;`

0 if inserted else schema->mem_usage()?

Reply:
How about modifying it in a separate PR? I just realized the logic, as well as the logic you mentioned in line 656 and line 957, may both be improper.

Reply:
ok